### PR TITLE
Canonical case of code and mandate type

### LINKED ISSUE #70: Canonicalise language, script, and region tags
Language, script, and region tags should be canonicalised, because
1. This matches how ECMA-402 works for other API.
1. ICU does this implicitly for some APIs resp. other ICU APIs require a canonicalised input to produce any result.

---

For example when `new Intl.DisplayNames("en", {type: "language"}).of("de-DD")` returns `"German (Germany)"`, for consistency we should then ensure that `new Intl.DisplayNames("en",{type: "region"}).of("DD")` returns `"Germany"`.

### PR DESCRIPTION
**Pull request by FrankYFTang**
Convert the code to canonical case
Mandate the "type" instead of default to "language"

close #70, #72, #73

@anba

### CODE REVIEW

**Comment by FrankYFTang** [comment]
@sffc @zbraniecki @leobalter @gibson042

**Review by zbraniecki — APPROVED** [review]
lgtm!

**Review by sffc — APPROVED** [review]

**Comment by FrankYFTang** [comment]
FYI v8 bug tracking https://bugs.chromium.org/p/v8/issues/detail?id=10623

File: spec.emu [hunk 1]
```diff
@@ -63,23 +63,28 @@ contributors: Google, Ecma International
...
+        The CanonicalCodeForDisplayNames abstract operation is called with arguments _type_, and _code_. It verifies that the _code_ argument represents a well-formed code according to the _type_ argument and returns the case-regularized form of the _code_. The algorithm refers to <a href="https://www.unicode.org/reports/tr35/#Identifiers">UTS 35's Unicode Language and Locale Identifiers grammar</a>. The following steps are taken:
       </p>
       <emu-alg>
         1. If _type_ is `"language"`, then
-          1. If _code_ matches the `unicode_language_id` production, return *true*.
-          1. Return *false*.
+          1. If _code_ does not matches the `unicode_language_id` production, throw a *RangeError* exceptipon.
+          1. Let _code_ be the result of mapping _code_ to lower case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
```
Comment (anba) [inline spec.emu:74]: `unicode_language_id` contains more than just `unicode_language_subtag`. Maybe it's best just to refer to <https://unicode.org/reports/tr35/tr35.html#Canonical_Unicode_Locale_Identifiers> here?

**Review by anba — COMMENTED** [review]
This doesn't seem to handle #70. AFAICS CLDR doesn't contain out-dated/historical names for languages/regions. so for example the region name for "DD" is automatically canonicalised to "DE", which then returns "Germany" instead of "East Germany".

File: spec.emu (hunk 1 above)
Comment (FrankYFTang) [reply spec.emu:74]: opps. forgot about that. Let me reopen #70

**Pull request merged**